soc/nxp/mcx/mcxnx4x/CMakeLists.txt

```diff
@@ -10,8 +10,11 @@ zephyr_compile_definitions_ifdef(CONFIG_NXP_LP_FLEXCOMM LPFLEXCOMM_INIT_NOT_USED
 
 zephyr_sources(
   soc.c
+  flash_clock_setup.c
```

Comment:
We should only need to relocate this code to RAM when `CONFIG_FLASH_MCUX_FLEXSPI_XIP` is set, right?

Reply:
Thank you for pointing this, I have updated the code.